daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/DataProcessor.scala

```diff
@@ -372,7 +372,7 @@ class DataProcessor private (
       validationMode match {
         case ValidationMode.Full | ValidationMode.Custom(_) =>
           val bos = new java.io.ByteArrayOutputStream()
-          val xmlOutputter = new XMLTextInfosetOutputter(bos, false)
+          val xmlOutputter = new XMLTextInfosetOutputter(bos, false, tunables.xmlOutputStyle)
```

Comment:
Yeah, I prefer we come up with something other than tunables for configuring these in the CLI, otherwise I think it becomes confusing when it is set.

In this particular case, this XMLTexInfosetOutputter should always have pretty-print safe disabled because this XMLTextInfostOutputter is only used for Xerces validation and the user will never see it and we don't have to worry about something reformatting and breaking the infoset.

Reply:
Hmmm. I see lots of need for future parameterization of infoset outputters/inputters. 

Is there a fundamental flaw with the way we do tunables now that you are advocating we move away from?

To me, whatever mechanism we provide for this is going to be substantially similar to the tunables object we already have. Or is there some different idea?

Plausible controls for infoset XML-related inputters/outputters are:
- substitution scheme to use for XML illegal characters
- pretty print safe/not
- threshold or guidance on when to use CDATA vs. escapifying
- control of when to output Unicode characters, and when to output `&#xHHHH;` notation and use Ascii as the charset. 
- Whether to append the XML "slug line" `<?xml .... >` at start of data
- Whether to append a byte-order-mark at start of data. 

I'm not suggesting we add these things now. Just that any/all of them are plausible things we'd want to implement by parameterizing infoset inputters/outputters with parameters settable/tunable from outside the DFDL schema.

Reply:
Ok. I get it. Tunables are for things created by the Compiler/ProcessorFactory/DataProcessor chain of stuff. 
So we should be able to just create a key-value thing, and pass an instance of it to any infoset inputter/outputter constructor, and then we have to provide some way to set those for CLI use, AND for TDML runner use (don't forget that case!)

Reply:
Good point, I hadn't thought about the TDML runner.

Though, one thing to consider is that the TDML runner is somewhat lax about what it compares. For example, I think elements look the same to TDML XML comparison regardless if CDATA is used or not.

Aslo, TDML runner uses all InfosetOutputters and compares the results. So, for example, if a TDML test configures an XMLTextInfosetOutputter to use UTF-8 but not other InfosetOutputters, then things might not compare the same.

These kinds of issues can probably be resolved, but we do also have the (now faster) CLI integration tests and API unit tests if it's difficult.

Reply:
@stevedlawrence can this conversation be marked resolved?